### ISSUE TITLE
the two way of nested values' mappedErrors are different?

### ISSUE THREAD

**Issue by alexgzhou** [ctavan/express-validator#165]
req.assert(['user', 'email'], 'not valid email').optional().isEmail();
req.assert('user.mobile', 'not valid mobile').optional().isMobilePhone('zh-CN');

In the previous express-validator version, if i use the two req.assert above to check the nested params, I got the same mappedErrors as mappedErrors['user.email'] and mappedErrors['user.mobile']; 
But now I get mappedErrors['user,email'] and mappedErrors['user.mobile'], **notice** the separator is a **comma** in the 'user,email', not a dot, is it on purpose?

And when I check three layers nested values like below, it does not work:
req.assert('user.personalInfo.idNo', 'not valid ID No.').optional().len(6, 20).isAlphanumeric();

And below is working, but again, I got a mappedError like mappedError['user,personalInfo.idNo'], **notice** the first separator is a **comma**, not a dot:
req.assert(['user','personalInfo.idNo'], 'not valid ID No.').optional().len(6, 20).isAlphanumeric();

Is it true that only two values nested is OK with the syntax of 'dot-notation'?

**Comment by rustybailey** [comment]
Thanks for the reporting this. What was the last version you used that worked?

**Comment by rustybailey** [comment]
@Jakeii or @chrissinclair: would either of you be able to look into this?

**Comment by rustybailey** [comment]
After doing some investigation, this errors seems to have occurred because of the giant code refactor a couple months ago. Essentially, in [v2.11.0](https://github.com/ctavan/express-validator/tree/v2.11.0) (before the refactor), if you ran the test `nestedInputTest.js`, the errors would output as:
``` js
[ { param: 'user.fields.email', msg: 'not empty', value: '' },
  { param: 'user.fields.email', msg: 'not empty', value: '' },
  { param: 'user.fields.email', msg: 'valid email required', value: '' } ]
```
However, in [v2.12.0](https://github.com/ctavan/express-validator/tree/v2.12.0) (after the refactor), they output as:
``` js
[ { param: [ 'user', 'fields', 'email' ],
    msg: 'not empty',
    value: '' },
  { param: 'user.fields.email', msg: 'not empty', value: '' },
  { param: [ 'user', 'fields', 'email' ],
    msg: 'valid email required',
    value: '' } ]
```
Notice how it doesn't convert `[ 'user', 'fields', 'email' ]` to `'user.fields.email'` in the param field.

So we need to:
1. Figure out what was missed in that refactor between the two releases.
2. Enhance the tests for `nestedInputTest.js` and `nestedInputSanitizeTest.js` to check the output of the errors to make sure the param string is in the correct format.

**Comment by chrissinclair** [comment]
Just to be clear on this - this is a regression issue and the correct behaviour is to output the param in the form `'user.fields.email'`?

It's a bank holiday here in the UK so I might be able to get some time and have a look.

**Comment by rustybailey** [comment]
Yes, this is a regression issue. Actually, I may be reaching a point where I almost have a PR ready, but if you have time would definitely appreciate you looking over the PR. I'll tag you in it when it's ready.

**Comment by rustybailey** [comment]
Pull request is open for this now and ready for review.

**Comment by rustybailey** [comment]
This should be fixed in v2.17.1. Please reopen if this is still occurring.

**Comment by alexgzhou** [comment]
Thanks guys, you awesome!
